dashboard/app/models/user.rb

```diff
@@ -830,6 +831,31 @@ def update_with_password(params, *options)
     end
   end
 
+  def update_primary_authentication_option(user: {email: nil, hashed_email: nil})
```

Comment:
When we add UI for options other than email, this will need to be updated, we'll probably want to think about how to best do this. I'm guessing at that point, we'll probably want to rename this so that it's an email-specific method, but for now this seems reasonable.